app/server/appsmith-plugins/dynamoPlugin/src/main/java/com/external/plugins/DynamoPlugin.java

```
@@ -0,0 +1,347 @@
+                requestClass = Class.forName("software.amazon.awssdk.services.dynamodb.model." + action + "Request");
+            } catch (ClassNotFoundException e) {
+                return Mono.error(new AppsmithPluginException(
+                        AppsmithPluginError.PLUGIN_ERROR,
+                        "Unknown action: `" + action + "`. Note that action names are case-sensitive."
+                ));
+            }
+
+            try {
+                final Method actionExecuteMethod = DynamoDbClient.class.getMethod(action.substring(0, 1).toLowerCase() + action.substring(1), requestClass);
+                final DynamoDbResponse response = (DynamoDbResponse) actionExecuteMethod.invoke(ddb, plainToSdk(parameters, requestClass));
+                result.setBody(sdkToPlain(response));
+            } catch (AppsmithPluginException | InvocationTargetException | IllegalAccessException
+                    | NoSuchMethodException | InstantiationException e) {
+                return Mono.error(e.getCause() == null ? e : e.getCause());
+            }
+
+            result.setIsExecutionSuccess(true);
+            return Mono.just(result);
+        }
+
+        @Override
+        public Mono<DynamoDbClient> datasourceCreate(DatasourceConfiguration datasourceConfiguration) {
+            final DynamoDbClientBuilder builder = DynamoDbClient.builder();
+
+            if (!CollectionUtils.isEmpty(datasourceConfiguration.getEndpoints())) {
+                final Endpoint endpoint = datasourceConfiguration.getEndpoints().get(0);
+                builder.endpointOverride(URI.create("http://" + endpoint.getHost() + ":" + endpoint.getPort()));
+            }
+
+            final AuthenticationDTO authentication = datasourceConfiguration.getAuthentication();
+            if (authentication != null) {
+                builder.region(authentication.getDatabaseName() != null
+                        ? Region.of(authentication.getDatabaseName()) : Region.AP_SOUTH_1);
```

Any particular reason we are defaulting to `AP_SOUTH_1` region ? I can see that in the `validateDatasource` function, we are checking for the region. Should we just error out here incase the region is still not set?

---

No reason really. Just the first one that showed up. Changed now to throw an error instead of having a default.